runtime/calamari/src/lib.rs

```diff
@@ -752,8 +780,48 @@ pub type Executive = frame_executive::Executive<
     frame_system::ChainContext<Runtime>,
     Runtime,
     AllPalletsReversedWithSystemFirst,
-    (),
+    UpgradeSessionKeys,
 >;
+// When this is removed, should also remove `OldSessionKeys`.
+pub struct UpgradeSessionKeys;
+impl frame_support::traits::OnRuntimeUpgrade for UpgradeSessionKeys {
+    fn on_runtime_upgrade() -> frame_support::weights::Weight {
+        use opaque::transform_session_keys;
+        // transform_session_keys runs translate() on NextKeys and on QueuedKeys which is (at worst or faster than) 1 read 1 write
+        let validator_set_len: u64 = Session::queued_keys().len().try_into().unwrap();
```

Comment:
Seems `usize` can `into` `u64`, not 100% sure. Perhaps I'm wrong.

Reply:
It can't. I tried :D